### PR TITLE
Browser backend initialization fix and file picker updates

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

1. Changed AppBuilder API:
```diff
public static class BrowserAppBuilder
{
-    AppBuilder SetupBrowserApp()
+    Task StartBrowserAppAsync(this AppBuilder builder, string mainDivId, BrowserPlatformOptions? options = null);
+    Task SetupBrowserAppAsync(this AppBuilder builder, BrowserPlatformOptions? options = null);
     AppBuilder UseBrowser(AppBuilder builder);
}

public static class BlazorAppBuilder
{
-    AppBuilder SetupWithSingleViewLifetime(this AppBuilder builder);
-    AppBuilder UseBlazor(this AppBuilder builder)
-    AppBuilder Configure<TApp>()
+    Task StartBlazorAppAsync(this AppBuilder builder, BrowserPlatformOptions? options = null);
}
```
Where new async methods will properly resolve Avalonia scripts loading instead of relying on preloading. Also, consistency.

3. Fixed initialization order in the blazor. AppBuilder should be set up in the Main method.
4. Fixed custom file extensions in the open file picker on the browser backend.
5. Used [polyfill](https://github.com/jimmywarting/native-file-system-adapter) for the open file and folder pickers. Save file picker technically can be supported with hacks, but not in this PR.
6. Removed unused CanOpenRead/CanOpenWrite, these properties always returned "true" in all cases. 
7. Renamed TryGetFileFromPath-like methods added in preview5 to end with Async suffix.
8. Added new public IStorageItem.TryGetLocalPath() method to simplify API usage on desktop.

## Breaking changes

Breaking comparing to previous preview versions. AppBuilder method were changed.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030038-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by jp2masa** [comment]
Suggestion: add `Async` suffix to the new async methods.

File: src/Browser/Avalonia.Browser/BrowserPlatformSettings.cs [hunk 1]
```diff
@@ -7,25 +8,44 @@ internal class BrowserPlatformSettings : DefaultPlatformSettings
...
             ContrastPreference = _isHighContrast ? ColorContrastPreference.High : ColorContrastPreference.NoPreference
         };
     }
+
+    private void EnsureBackend()
+    {
+        if (!_isInitialized)
+        {
+            // WASM module has async nature of initialization. We can't native code right away during components registration. 
+            _isInitialized = true;
```
Comment (maxkatz6) [inline src/Browser/Avalonia.Browser/BrowserPlatformSettings.cs:39]: Technically, this change isn't required anymore after adding async Setup/Start methods. It didn't work properly anyway with sync initialization.

**Review by kekekeks — APPROVED** [review]
LGTM

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030095-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**